internal/config/global.go

```diff
@@ -1,91 +1,141 @@
 package config
 
 import (
-	"flag"
 	"fmt"
 	"net/url"
-	"os"
+	"regexp"
+	"strings"
 
 	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var (
+	supportedSchemes = regexp.MustCompile("^https?$")
 )
 
 type GlobalOptions struct {
-	OidcIssuerUrl,
-	OidcClientId,
 	FulcioUrl,
 	RekorUrl,
+	OidcIssuerUrl *UrlValue
+
+	OidcClientId,
 	IdToken,
 	ArtifactUri string
-	ArtifactDigest *Digest
+	ArtifactDigest *DigestValue
 }
 
-func NewGlobalOptions() GlobalOptions {
-	return GlobalOptions{
-		FulcioUrl:      "https://fulcio.sigstore.dev",
-		RekorUrl:       "https://rekor.sigstore.dev",
-		OidcIssuerUrl:  "https://oauth2.sigstore.dev/auth",
-		ArtifactDigest: &Digest{},
+var (
+	_ pflag.Value = &UrlValue{}
```

Comment:
Can you explain what's happening here?

Reply:
Cobra supports custom flag types if the type implements the `pflag.Value` interface. I made a custom type for the URLs and digest in order to have Cobra do validation of the flag value. 

These two lines are checking that it meets the `pflag.Value` interface. It was useful for scaffolding the implementation, but probably not needed now.